app/models/targeting.rb

```diff
@@ -0,0 +1,28 @@
+class Targeting < ActiveRecord::Base
+
+  belongs_to :user
+  belongs_to :bookmark
+
+  has_many :targeting_hosts, :dependent => :destroy
+  has_many :hosts, :through => :targeting_hosts
+  has_many :template_invocations, :dependent => :destroy
+
+  validates :bookmark_id, :presence => {:message => N_("must be specified if search query is blank")},
+            :if => Proc.new { |a| a.search_query.blank? }
```

Comment:
maybe it makes sense to always create a bookmark, and flag it maniacally, vs storing a search_query? this will allow to give a user a history of searches with possible dates? (maybe save the last used time, and how many times it was used)